tests/test_modeling_tf_common.py

```diff
@@ -597,9 +606,15 @@ def test_loss_computation(self):
                 added_label = prepared_for_class[list(prepared_for_class.keys() - inputs_dict.keys())[0]]
                 loss_size = tf.size(added_label)
 
+                if model.__class__ in TF_MODEL_FOR_CAUSAL_LM_MAPPING.values():
+                    # if loss is causal lm loss, labels are shift, so that one label per batch
+                    # is cut
+                    loss_size = loss_size - self.model_tester.batch_size
```

Comment:
This is a bit hacky, but I don't see another way around it at the moment @sgugger @jplu - CLM loss shifts the tokens and thus cuts off one token

Reply:
For me it is ok to do like this, it doesn't seems too odd.